docs/src/cli/conventions.md

```diff
@@ -46,15 +46,15 @@ on your wallet type.
 #### Paper Wallet
 
 In a paper wallet, the keypair is securely derived from the seed words and
-optional passphrase you entered when the wallet was create. To use a paper wallet
-keypair anywhere the `<KEYPAIR>` text is shown in examples or help documents,
-enter the word `ASK` and the program will prompt you to enter your seed words
-when you run the command.
+optional passphrase you entered when the wallet was create. To use a paper
+wallet keypair anywhere the `<KEYPAIR>` text is shown in examples or help
+documents, enter the uri scheme `ask://` and the program will prompt you to
+enter your seed words when you run the command.
 
 To display the wallet address of a Paper Wallet:
 
 ```bash
-solana-keygen pubkey ASK
+solana-keygen pubkey ask://
```

Comment:
Do these generate the same address?  At a glance, I'd guess that `ask://` would generate the key at `ASK/44'/501'`.

Reply:
Yes, these do generate the same address. Have you read this section yet? https://github.com/solana-labs/solana/pull/16942/files#diff-74038ef7d737ad7ebedf6ddb1f940e6f15337de6cba7d331a3621cccdd81cd3aR123
Definitely would love suggestions to make docs more clear.

Reply:
I'm confused, do I have to write `ask://?full-path=""`?  As I recall, `usb://` gets the Solana-specific address at `44'/501'`. I don't see what `ask://` wouldn't do the same (which is different than what `ASK` would have output).

Reply:
No, you would write `ask://?key=m` to use solana's bip44 derivation path root, `44'/501'`. 
Now that we're depending on the `derivation-path` crate, `usb://ledger?key=m` actually also derives the solana bip44 root, although of course it's easier to just not supply the query string at all.

I can see the argument for making `ask://` no query behave the same as `usb://` no query, but I was concerned about retaining compatibility for current seed phrase users. They would otherwise have to continue using `ASK` or we would have to build in a special query string for `ask://`, since the current output of `ASK` is a key that is not anywhere in the bip32 derivation.
 cc @t-nelson

Reply:
We're going to work toward documenting everything in URI form going forward. `ask:` (I'm somewhat dubious as to whether we should support the network prefix `://`) is a direct replacement for `ASK`, which should be considered deprecated.  `usb://` is overloaded in that it implicitly derives BIP44 keypairs.  This is the bug IMO, but we keep the special-case for backward compatibility.  So: `ask:` gives the raw ed25519 keypair, `ask:?key=X[/Y]` gives the BIP44 account[/change] address, `ask:?full-path=m/X[/Y[/Z[/...]]]` gives the corresponding BIP32 address.  This is consistent across all other schemes

Reply:
How about:
* `ask://` means `ASK`, does not accept a query string, and does not imply any key derivation
* `ask://solana` is an alias for `ask://?prefix=44'/501'`
* `usb://ledger` is an alias for `usb://ledger?prefix=44'/501'`

Also, how about renaming `ask://` to `seedphrase://` or `stdin://seedphrase`?

Reply:
@CriesofCarrots, wdyt?

Reply:
@garious I rather like `stdin://seedphrase`. We do already have a `stdin` scheme implementation for piping in json keypair (same as passing `-`). What host do you like for that?

For `seedphrase`, I would envision:
- `stdin://seedphrase` => `ASK` (raw ed25519 keypair)
- `stdin://seedphrase?key=1/2` => solana bip44 `m/44/501/1/2`
- `stdin://seedphrase?full-path=m/44/2017/1/2`

Or would you want to use the `solana` alias as well?

Reply:
Hmm, I don't care for 'full-path' because it conflicts with 'key' (unless 'key' always appends, in which case 'full-path' is not the "full" path). A 'prefix' parameter that defaults to "m/44'/501'" seems more intuitive to me. To get ASK, you can then write "stdin://seedphrase?prefix=''"